usaspending_api/recipient/v2/views/states.py

```diff
@@ -112,15 +111,29 @@ def get_all_states(year=None, award_type_codes=None, subawards=False):
             .values("pop_state_code", "total", "distinct_awards", "outlay_total")
         )
 
-        results = [
-            {
-                "pop_state_code": row["pop_state_code"],
-                "total": row["total"],
-                "count": len(set(row["distinct_awards"].split(","))),
-                "total_outlays": row["outlay_total"],
-            }
-            for row in list(queryset)
-        ]
+        if len(queryset) > 0:
+            results = [
+                {
+                    "pop_state_code": row["pop_state_code"],
+                    "total": row["total"],
+                    "count": len(set(row["distinct_awards"].split(","))),
+                    "total_outlays": row["outlay_total"],
+                }
+                for row in list(queryset)
+            ]
+        # In the case where there were no results, return a list of state codes with 0 for the amounts
+        # This is a common case on the first day of a new fiscal year
+        else:
+            queryset = SummaryStateView.objects.all().distinct("pop_state_code").values("pop_state_code")
```

Comment:
Probably not much of a concern since the two queries this will run against the DB are relatively short. However, if there are any performance concerns you could also query one with a LEFT OUTER JOIN between SummaryStateView that returns all unique states and then SummaryStateView that captures the values per state. Then in the response set values to their default.